### PR TITLE
jwt-cpp: add v0.7.0

### PR DESCRIPTION
**Pull request by prince-chrismc**

### CODE REVIEW

File: var/spack/repos/builtin/packages/jwt-cpp/package.py [hunk 1]
```diff
@@ -16,12 +16,15 @@ class JwtCpp(CMakePackage):
 
     license("MIT")
 
+    version("0.6.0", sha256="b9eb270e3ba8221e4b2bc38723c9a1cb4fa6c241a42908b9a334daff31137406")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/jwt-cpp/package.py]: Did you mean
```suggestion
    version("0.7.0", sha256="b9eb270e3ba8221e4b2bc38723c9a1cb4fa6c241a42908b9a334daff31137406")
```

Comment (prince-chrismc) [reply var/spack/repos/builtin/packages/jwt-cpp/package.py]: Yes 😮‍💨 bad copy paste 🤐

**Comment by wdconinc** [comment]
There seems to be a new dependency on `nlohmann_json`, with implicit fallback to FetchContent if not found (which bumps the minimum cmake version). Generally spack avoids implicit FetchContent.

**Comment by prince-chrismc** [comment]
"New" its was always supported, it's only used to build the examples and its not installed. In 0.6.0 it's was just embedded https://github.com/Thalhammer/jwt-cpp/tree/v0.6.0/include/nlohmann,  this is still a problem in for picojson

(I've never used spack) The version logic for cmake args is very confusing so I am not sure if disabling that is enough?

**Comment by wdconinc** [comment]
The issue with nlohmann_json is that this upgrade would break airgapped installs (and I do see the possible irony of wanting "web tokens" to work on airgapped systems...). Previously (0.6.0) the `json.hpp` is included in the repo and so the distribution is fully vendored. Now (0.7.0) it would try to FetchContent nlohmann_json, which would fail on airgapped systems. 

But I'm more confused: even on non-airgapped systems without nlohmann_json, 0.7.0 installs `include/jwt-cpp/traits/nlohmann-json/traits.h` which `#include`s `nlohmann/jsoh.hpp` even though FetchContent doesn't install it. That seems like it wouldn't work downstream (or, worse, it could pick up a different nlohmann_json version when a downstream project *does* use a system nlohmann_json).

I think it's fairly easy here to just add the following to the PR to ensure nlohmann_json is installed first:
```
    depends_on("nlohmann-json", when="@0.7.0:")
```
That seems to do the job in my tests.

**Review by wdconinc — APPROVED** [review]
Merging develop into this branch changed it into a comment change only. I think there is value in the comment (similar to a comment right below it), so I'm approving it. Thanks @prince-chrismc! At this time there is still no picojson in spack.

**Comment by prince-chrismc** [comment]
Ohh it seems another PR Merged the original changes 🤔 

I do agree the behavior is not ideal. There should be an optional combo for selecting the JSON library. However making this consistent for CMake and multiple packages managers is painful... OpenSSL is different for conan/vcpkg/nuget and  each ecosystem has its own preception of what's wrong.

The intention is to have a default opt-out JSON library. In master nholmann is only used for testing. Users regrettably need to explicitly pick the json library

**Pull request merged**